docs: add examples for Select overlay config options (#3563)
vaadin/docs: articles/components/select/index.adoc

[CAUTION]
Some assistive technologies might not announce disabled options.

== Overlay

> Select provides the following API to customize the overlay where the list of options is rendered.

Reviewer: ```suggestion
You can customize the position and the width of the Select overlay, where the list of options is rendered. You can also add CSS class names to the overlay to customize its styles.
```

[role="since:com.vaadin:vaadin@V24.5"]
=== Overlay Position

> By default, Select overlay overlaps the input-field when opened. In order to change this and make the drop-down menu look more like ComboBox, use `noVerticalOverlap` API.

Reviewer 2: ```suggestion
By default, the drop-down overlaps the field when opened. The `noVerticalOverlap` API can be used to move the drop-down below the field instead.
```

> By default, the drop-down overlaps the field when opened. There is an API for configuring Select to place the overlay below the field (or above it, if there is not enough space at the bottom).

Reviewer: ```suggestion
The overlay is shown on top of the input field by default. You can configure the placement so that it doesn't cover the input field.
```

[.example]
--
ifdef::lit[]
[source,typescript]
----
include::{root}/frontend/demo/component/select/select-no-vertical-overlap.ts[render,tags=snippet,indent=0,group=Lit]
----
endif::[]

ifdef::flow[]
[source,java]
----
include::{root}/src/main/java/com/vaadin/demo/component/select/SelectNoVerticalOverlap.java[render,tags=snippet,indent=0,group=Flow]
----
endif::[]

ifdef::react[]
[source,tsx]
----
include::{root}/frontend/demo/component/select/react/select-no-vertical-overlap.tsx[render,tags=snippet,indent=0,group=React]
----
endif::[]
--

[role="since:com.vaadin:vaadin@V24.5"]
=== Overlay Width

> Use the [methodname]`setOverlayWidth()` API in Flow or `--vaadin-select-overlay-width` style property on the web component to customize the overlay width.

Reviewer 3: nit: we don't typically include API specifics in the documentation. Users can find the technical _how_ in the code examples. One reason is that any outdated code in the Asciidoc wouldn't be caught by the compiler.

Author: Updated new paragraphs to not include API specifics and use a better wording instead.

> [.example]

Reviewer 3: Is it intentional not to have live examples for these?

Author: We actually have live example for ComboBox popup width. I'll add similar one here.

Author: Added a demo.

> Use the [methodname]`setOverlayWidth()` Java API or `--vaadin-select-overlay-width` custom CSS property on the web component to customize the overlay width.

Reviewer 2: ```suggestion
Use the [methodname]`setOverlayWidth()` API in Flow or `--vaadin-select-overlay-width` style property on the web component to customize the overlay width.
```

> The width of the overlay is, by default, the same width as the input field. The overlay width can be overridden to any fixed width in cases where the default width is too narrow.

Reviewer: The overlay in Select can grow wider than the input field. Not sure if this is the clearest way to describe the behavior.

```suggestion
The overlay is as wide as the options in it and at least as wide as the input field. The overlay width can be overridden to any fixed width in cases where the default width is too narrow.
```

--
[source,java]
----
<source-info group="Flow"></source-info>
Select<String> select = new Select<>();
select.setOverlayWidth(350, Unit.PIXELS);
----
[source,tsx]
----
<source-info group="React"></source-info>
<Select style={{ '--vaadin-select-overlay-width': '350px' }} />
----
[source,html]
----
<source-info group="Lit"></source-info>
<vaadin-select style="--vaadin-select-overlay-width: 350px"></vaadin-select>
----
--

=== Overlay Classname

> Use the [methodname]`addOverlayClassName()` Java API or `overlayClass` web component property to set CSS class names on the select overlay for styling.

Reviewer 2: ```suggestion
Use the [methodname]`addOverlayClassName()` API in Flow or the `overlayClass` web component property to set a CSS class name on the overlay for styling.
```
